### PR TITLE
feat: fixed auth

### PR DESCRIPTION
**Pull request by ablevargheseme**

### CODE REVIEW

File: src/backend/api/handler/proof-of-work/index.ts [hunk 1]
```diff
@@ -13,10 +15,23 @@ const getAllProofOfWork = async (req: NextApiRequest, res: NextApiResponse) => {
 
 const createProofOfWork = async (req: NextApiRequest, res: NextApiResponse) => {
   const { body } = req;
+  const acessTokenRx = req.body.acessToken;
+  const walletAddressRx = req.body.walletAddress;
```
Comment (sreejinsreenivasan) [inline src/backend/api/handler/proof-of-work/index.ts:19]: can we move this validation logic into a separate function so that we can re-use in other routes?

File: src/backend/api/handler/proof-of-work/index.ts [hunk 2]
```diff
@@ -13,10 +15,23 @@ const getAllProofOfWork = async (req: NextApiRequest, res: NextApiResponse) => {
...
+  const jwtDecoded = JSON.parse(
+    Buffer.from(acessTokenRx.split(".")[1], "base64").toString()
+  );
+  const decodedWalletAddress = jwtDecoded.id;
+  console.log("walletaddressRx", walletAddressRx);
```
Comment (sreejinsreenivasan) [inline src/backend/api/handler/proof-of-work/index.ts:24]: Please remove un unnecessary console.log's. we need to logs to track the code execution but we don't want to sensitive info like access token

File: src/components/profile/loggedInUser.tsx [hunk 3]
```diff
@@ -1,23 +1,62 @@
...
+import useLensUser from "../../lib/auth/useLensUser";
+import useLogin from "../../lib/auth/useLogin";
 import { ProfileComponent } from ".";
 import { Main } from "../main";
 
+
+
 interface LoggedInUserProps {
-  address: string;
+  propsaddress: string;
```
Comment (sreejinsreenivasan) [inline src/components/profile/loggedInUser.tsx:16]: `propsaddress` the naming seems ambiguous, because we use props mechanism to pass data to child components, we dont need to mention it is props value exclusively.

I recommend to keep it in old way.

File: src/components/profile/loggedInUser.tsx [hunk 4]
```diff
@@ -1,23 +1,62 @@
...
-  const { data: profile } = useDefaultProfileQuery({
-    request: {
-      // ethereumAddress: "0xccd37217ed186b094469FA90b375BE23757609c8",
-      ethereumAddress: address,
-      // TODO: use the user's ethereum address. the hard coded address is for testing purposes only
-    },
-  });
+const LoggedInUser = ({ propsaddress }: LoggedInUserProps) => {
+
+  const address = useAddress(); // Detect the connected address
```
Comment (sreejinsreenivasan) [inline src/components/profile/loggedInUser.tsx:21]: this is an extra call to get user's connected wallet address, which we are already receiving as props.

File: src/components/profile/loggedInUser.tsx [hunk 5]
```diff
@@ -1,23 +1,62 @@
...
+  const isOnWrongNetwork = useNetworkMismatch(); // Detect if the user is on the wrong network
+  const [, switchNetwork] = useNetwork(); // Function to switch the network.
+  const { isSignedInQuery, profileQuery } = useLensUser();
+  const { mutate: requestLogin } = useLogin();
+
+  // 1. User needs to connect their wallet
```
Comment (sreejinsreenivasan) [inline src/components/profile/loggedInUser.tsx:27]: can we move this entire login logic someplace else? because we might need to use this on other pages as well.
Maybe in the `<ConnectComponent/>`? and pass a path as props after a successful login
Maybe we should keep a login state in global context

File: src/graphql/auth-fetcher.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-const endpoint = process.env.NEXT_PUBLIC_LENS_API_URI;
+const endpoint = "https://api-mumbai.lens.dev";
```
Comment (sreejinsreenivasan) [inline src/graphql/auth-fetcher.ts:1]: This seems like an unnecessary change. I put this inside a `.env` file because we need the ability to switch between the `main` net and the `test` net in different environments.

If you did this because of any certain use case please advocate it.

**Review by sreejinsreenivasan — COMMENTED** [review]
The code looks fine. please address the reviews.
Please remove un unnecessary `console.log`'s.

File: src/components/profile/loggedInUser.tsx (hunk 3 above)
Comment (ablevargheseme) [reply src/components/profile/loggedInUser.tsx:16]: yaaa, we can keep it in the old way, is there any issue with taking the address from hook, without using the props address

File: src/backend/api/handler/proof-of-work/index.ts (hunk 2 above)
Comment (ablevargheseme) [reply src/backend/api/handler/proof-of-work/index.ts:24]: ok

File: src/backend/api/handler/proof-of-work/index.ts (hunk 1 above)
Comment (ablevargheseme) [reply src/backend/api/handler/proof-of-work/index.ts:19]: hmmm, yaaa good idea, we can use it for endorsement

File: src/components/profile/loggedInUser.tsx (hunk 4 above)
Comment (ablevargheseme) [reply src/components/profile/loggedInUser.tsx:21]: ok

File: src/graphql/auth-fetcher.ts (hunk 6 above)
Comment (ablevargheseme) [reply src/graphql/auth-fetcher.ts:1]: no certain  use, let us set it in .env

File: src/components/profile/loggedInUser.tsx (hunk 5 above)
Comment (ablevargheseme) [reply src/components/profile/loggedInUser.tsx:27]: ok,

File: src/components/profile/loggedInUser.tsx (hunk 3 above)
Comment (sreejinsreenivasan) [reply src/components/profile/loggedInUser.tsx:16]: No issues, if the data is passing in the props every component will have same state. But if we access the address using the `useAddress` hook individually in multiple places it can issues.
